### ISSUE TITLE
Bug: find_arrow_all in version 1.0.1 returns wrong schema for nested bson.ObjectId while bson.ObjectId on root level works as documented

### ISSUE THREAD

**Issue by sibbiii** [mongodb-labs/mongo-arrow#163]
Hi,

Let me thank you for the new 1.0.1 release and making the bson.ObjectId an Extension Type. 
This new feature seems to work for all keys on the root level, but not for the ones nested within Objects.

Steps to reproduce:

```    
    collection = ...
    obj_id = bson.ObjectId()

    collection.insert_one(
        {'_id': obj_id,
         'id1': obj_id,
         'obj': {'id2': obj_id}
         })

    print(pymongoarrow.api.find_arrow_all(collection, {}))
    print(pymongoarrow.api.find_arrow_all(collection, {}).to_pandas())
```

    
The output is:

```   
    _id: extension<arrow.py_extension_type<ObjectIdType>>
    id1: extension<arrow.py_extension_type<ObjectIdType>>
    obj: struct<id2: fixed_size_binary[12]>
           child 0, id2: fixed_size_binary[12]
```
```  
                        _id                       id1                                            obj
0  649c7842aa528cb1069843d2  649c7842aa528cb1069843d2  {'id2': b'd\x9cxB\xaaR\x8c\xb1\x06\x98C\xd2'}
```  

Note that id1 is of type arrow.py_extension_type<ObjectIdType> and id2 of type fixed_size_binary[12]. It seems that nested ObjectIDs are treated as before version 1.0.0 so maybe something has been overlooked here at the latest upgrade.

Defining the schema manually leads to the same output.

Thanks,
Sebastian

**Comment by blink1073** [comment]
Thanks for reporting!  I opened https://jira.mongodb.org/browse/ARROW-176 to track.

**Comment by blink1073** [comment]
Fixed by #166
